transcom/mymove
# INT - B-21481 -  DB -  Add HI/AK duty locations

## [B-21481](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1016235&RoomContext=TeamRoom%3A855664&concept=TeamRoom)

## Summary

This PR introduces a new database migration file to insert 533 Hawaii and Alaska duty locations. A total of 533 duty location records were created out of possible 639 from the provided TRDM file. 106 duty locations were ignored/excluded per TRDM's instructions(flag).   

Feature flag -- enable_alaska -  has been been incorporated to include/exclude both HI and AK duty locations from duty location search results. This feature serves as a transitional feature to bridge the work between data migration and the issue related to empty counseling office dropdown. Another backlog ticket will be created to address the counseling office issue. This FF enables this ticket to be fulfilled per AC.

Note: This PR contains [DB - INT PR 21442](https://github.com/transcom/mymove/pull/13872)

**New Migration File: migrations/app/schema/20241017183144_add_AK_HI_duty_locations.up.sql**

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Edit .envrc and set FEATURE_FLAG_ENABLE_ALASKA to TRUE. Run direnv allow. Login as customer/office. Goto view containing Duty Location widget and search for Alaska and Hawaii locations....ie Honolulu, Anchorage ...etc... See attached file below for 599 expected locations. Confirm AK and HI locations are present in the results.
2. Edit .envrc and set FEATURE_FLAG_ENABLE_ALASKA to FALSE. Run direnv allow. Login as customer/office. Goto view containing Duty Location widget and search for Alaska and Hawaii locations....ie Honolulu, Anchorage ...etc... See attached file below for 599 expected locations. Confirm AK and HI locations are not present in the results.

**599 Duty Locations:**
[HI-AK-data-1729264016473.csv](https://github.com/user-attachments/files/17462373/HI-AK-data-1729264016473.csv)

Note: 
106 Excluded Duty Locations:
[null-us_post_region_cities_id.csv](https://github.com/user-attachments/files/17462406/null-us_post_region_cities_id.csv)


### Database

#### Any new migrations/schema changes:


## Screenshots
